prow/cluster/monitoring/prow_servicemonitors.yaml

```diff
@@ -169,3 +169,22 @@ spec:
   selector:
     matchLabels:
       app: tide
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app: boskos
+  name: boskos
+  namespace: prow-monitoring
+spec:
+  endpoints:
+    - interval: 30s
+      port: default
+      scheme: http
+  namespaceSelector:
+    matchNames:
+      - test-pods
```

Comment:
Isn't `boskos-metrics` running in the build cluster, not the service cluster?

Reply:
Good question.

Assuming that is the case, is there currently a way to define talking to an external service/cluster using this new monitoring framework? The docs are currently limited to the use of a ServiceMonitor.

Reply:
You could have a `Service` in the service cluster that ran nginx or whatever to forward to the other location ...

Reply:
That's kind of a hack though -- I think preferably we would deploy prometheus on the build cluster and scrape there, then configure that as a backend for grafana to pull from

Reply:
We will likely need to use [auth](https://grafana.com/docs/plugins/developing/auth-for-datasources/) on that link

Reply:
Yes, boskos is running in the test-pods namespace in different k8s cluster than the rest of prow...

Reply:
Prometheus itself does not [provide auth](https://prometheus.io/docs/guides/basic-auth/).
I feel an auth proxy like nginx is kind of unavoidable.
Prometheus and app staying in the same cluster sounds good to me and use grafana to join the dashboards.